src/account/account.cairo

```diff
@@ -71,10 +67,8 @@ mod AccountComponent {
 
             // Check tx version
             let tx_info = get_tx_info().unbox();
-            let version = tx_info.version;
-            if version != TRANSACTION_VERSION {
-                assert(version == QUERY_VERSION, Errors::INVALID_TX_VERSION);
-            }
+            let tx_version: u256 = tx_info.version.into();
```

Comment:
you mentioned this was more efficient than casting to smaller uint values, do i remember correctly?

Reply:
Not exactly. Setting the `TRANSACTION_VERSION` const as a u256 was more efficient in the above implementation

To your point, casting to smaller uints is indeed even more efficient, but the value we get from `tx_info` is a felt252 and we'd have to include a conditional for `QUERY_VERSION` because it's 2**128. If we're comfortable with using a smaller uint, how small is too small? Casting to u128 saves 980 gas and casting to u64 saves 1080

Reply:
I changed the version to u64. There's no further efficiency in going smaller. Let me know what you think

Reply:
sounds good. let's document the rationale for our choice